nixos/modules/programs/qgroundcontrol.nix

```diff
@@ -40,12 +40,14 @@ in
       '';
     };
 
-    # Security wrapper
+    # Security wrapper: use securitywrap to set the real group ID to tty so that gtk doesn't crash
     security.wrappers.qgroundcontrol = {
-      source = lib.getExe cfg.package;
-      owner = "root"; # Sensible default; not setuid so this is not a security risk
-      group = "tty";
-      setgid = true;
+      source = pkgs.writeShellScript "qgroundcontrol-securitywrapper" ''
+        exec ${lib.getExe pkgs.securitywrap} --reset-uid --set-real-gid dialout ${lib.getExe cfg.package} $@
+      '';
```

Comment:
We really need a new program for that and cannot fix this otherwise?

Reply:
Unfortunately, it looks like it.